subxt/src/rpc/lightclient/background.rs

```diff
@@ -0,0 +1,425 @@
+                    if sender.send(Ok(result)).is_err() {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    }
+                } else if let Some((sub_id_sender, sender)) = self.id_to_subscription.remove(&id) {
+                    let mut sub_id = result.to_string();
+                    sub_id.retain(|ch| ch.is_ascii_digit());
+                    tracing::trace!(target: LOG_TARGET, "Received subscription ID: {}", sub_id);
+
+                    if sub_id_sender.send(Ok(result)).is_err() {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    } else {
+                        // Track this subscription ID if send is successful.
+                        self.subscriptions.insert(sub_id, sender);
+                    }
+                }
+            }
+            Ok(RpcResponse::Subscription { method, id, result }) => {
+                // Subxt calls into `author_submitAndWatchExtrinsic`, however the smoldot produces
+                // `{"event":"broadcasted","numPeers":1}` and `{"event":"validated"}` which are part
+                // of the RPC V2 API. Ignore those events.
+                if method == "transaction_unstable_watchEvent"
+                    && (result.to_string().contains("broadcasted")
+                        || result.to_string().contains("validated"))
+                {
+                    tracing::debug!(target: LOG_TARGET, "Ignoring notification {:?}", result);
+                    return;
+                }
+
+                if let Some(sender) = self.subscriptions.get_mut(&id) {
+                    // Send the current notification response.
+                    if sender.send(result).await.is_err() {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            "Cannot send notification to subscription {:?}",
+                            id
+                        );
+
+                        // Remove the sender if the subscription dropped the receiver.
+                        self.subscriptions.remove(&id);
+                    }
+                }
+            }
+            Err(err) => {
+                tracing::warn!(target: LOG_TARGET, "cannot decode RPC response {:?}", err);
+            }
+        }
+    }
+
+    /// Perform the main background task:
+    /// - receiving user registration for RPC method / subscriptions
+    /// - providing the results from the light client back to users.
+    pub async fn start_task(
+        &mut self,
+        backend: mpsc::Receiver<BackendMessage>,
+        rpc_responses: smoldot_light::JsonRpcResponses,
```

Comment:
Nit: The word `backend` for "messages coming from subxt" confuses me a little, because I see these messages as coming from the "frontend" or user if anything; for me, something like `from_subxt` and `from_node` might be clearer :)